include/xtensor/xstridedview.hpp

```diff
@@ -0,0 +1,567 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille and Sylvain Corlay                     *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XSTRIDEDVIEW_HPP
+#define XSTRIDEDVIEW_HPP
+
+#include <algorithm>
+#include <cstddef>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include "xexpression.hpp"
+#include "xiterable.hpp"
+#include "xstrides.hpp"
+#include "xutils.hpp"
+
+namespace xt
+{
+    template <class CT, class CD>
+    class xstrided_view;
+
+    template <class CT, class CD>
+    struct xcontainer_inner_types<xstrided_view<CT, CD>>
+    {
+        using xexpression_type = std::decay_t<CT>;
+        using temporary_type = xarray<typename xexpression_type::value_type>;
+    };
+
+    template <class CT, class CD>
+    struct xiterable_inner_types<xstrided_view<CT, CD>>
+    {
+        using inner_shape_type = typename std::decay_t<CT>::shape_type;
+        using const_stepper = xindexed_stepper<xstrided_view<CT, CD>>;
+        using stepper = xindexed_stepper<xstrided_view<CT, CD>, false>;
+        using const_broadcast_iterator = xiterator<const_stepper, inner_shape_type*>;
+        using broadcast_iterator = xiterator<stepper, inner_shape_type*>;
+        using const_iterator = const_broadcast_iterator;
+        using iterator = broadcast_iterator;
+    };
+
+    /*****************
+     * xstrided_view *
+     *****************/
+
+    /**
+     * @class xstrided_view
+     * @brief View of an xexpression using strides
+     *
+     * The xstrided_view class implements a view utilizing an offset and strides 
+     * into a multidimensional xcontainer. The xstridedview is currently used 
+     * to implement `transpose`.
+     * @tparam CT the closure type of the \ref xexpression type underlying this view
+     * @tparam CD the closure type of the underlying data container
+     * 
+     * @sa stridedview, transpose
+     */
+    template <class CT, class CD>
```

Comment:
I added the `CD` template parameter, so later we can plug a "data shim" into the strided view so that it can be used with arbitrary xexpressions.